helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1391,6 +1393,8 @@ protected void checkLiveInstancesObservation(List<LiveInstance> liveInstances,
             manager.removeListener(keyBuilder.messages(instance), this);
             // remove customized state root listener for disconnected instances
             manager.removeListener(keyBuilder.customizedStatesRoot(instance), this);
+            // delete the customized state root path for disconnected instances
+            manager.deleteCustomizedStateRootForInstance(instance);
```

Comment:
Have a couple of the questions regarding this change:
1. Is there retry before marking a participant node as down? What are those config values for Venice?
2. My main concern is false negative (a node is mistakenly marked as down); so after a participant node, which used to be live, rejoins the cluster, Helix controller will resend state transition messages (offline->standby) to rebootstrap existing resources in the node, right?